lib/order.rb

```diff
@@ -1,18 +1,90 @@
+require 'csv'
+require 'awesome_print'
+
+
 module Grocery
   class Order
     attr_reader :id, :products
 
+
     def initialize(id, products)
       @id = id
       @products = products
     end
 
-    def total
-      # TODO: implement total
+
+    def self.all
+      orders = []
+      # "This is a class method."
+      # This is a class method to access the class variabel
+      #returns the csv file to an array with headers
+      #orders_csv = CSV.read('support/orders.csv', 'r')
+      orders_csv = CSV.read("support/orders.csv", 'r',headers: true).to_a
+
+
+      orders_csv.each do |line|
+        orders << Grocery::Order.new(*line)
+      end
+      return orders
+    end
+
+
+    def self.find(id)
+      a = self.all
+
+      if id > a.length
+        return nil
+      end
+
+      if id >=1
+        specific_order = a[id-1]
+      elsif id < 0
+        specific_order = a[id]
+      else
+        return nil
+      end
+      return specific_order
+    end
+
+
+
+
+    def self.total
+      a = self.all
+      sum = 0
```

Comment:
Total is supposed to be an instance method (call it on a single `Order`), not a class method.